### PR TITLE
feat: add key_exists_as_boundary_in_proof

### PR DESCRIPTION
**Pull request by QuantumExplorer**
## Summary

- Adds `key_exists_as_boundary_in_proof` to check whether a key exists as a `KVDigest` or `KVDigestCount` boundary node in proof bytes
- Useful for exclusive range queries (e.g. `RangeAfter(10)`) where the boundary key anchors the range in the proof but is not part of the result set
- Merk level: scans decoded proof ops for matching boundary nodes
- GroveDB level: decodes `GroveDBProof`, navigates to the correct layer by path, delegates to merk function (supports V0 and V1 proofs)

## Test plan

- [x] `RangeAfter` — boundary key detected at exclusive start
- [x] `RangeAfterTo` — boundary key detected
- [x] `RangeAfterToInclusive` — boundary key detected
- [x] Non-existent key not reported as boundary
- [x] Inclusive `Range` start key is a result, not a boundary
- [x] Full GroveDB round-trip: insert, prove, verify, check boundary

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added boundary key verification capability for proofs, enabling validation of whether specific keys exist as boundaries in proof structures across various range query scenarios.

* **Tests**
  * Expanded test coverage for boundary key verification across multiple query types and edge cases, including inclusive/exclusive range combinations and non-existent key handling.

### CODE REVIEW

File: merk/src/proofs/query/verify.rs [hunk 1]
```diff
@@ -673,3 +673,29 @@ impl fmt::Display for ProofVerificationResult {
...
+            | Op::Push(Node::KVDigestCount(k, _, _))
+            | Op::PushInverted(Node::KVDigestCount(k, _, _)) => {
+                if k.as_slice() == key {
+                    return Ok(true);
+                }
+            }
+            _ => {}
+        }
+    }
+    Ok(false)
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply merk/src/proofs/query/verify.rs:700]: it's fine

**Review by QuantumExplorer — COMMENTED** [review]
Self Reviewed

File: merk/src/proofs/query/verify.rs [hunk 2]
```diff
@@ -673,3 +673,29 @@ impl fmt::Display for ProofVerificationResult {
...
+/// bytes. A boundary element is a `KVDigest` or `KVDigestCount` node — it
+/// proves the key exists in the tree without revealing the value.
+///
+/// This is useful for exclusive range queries (e.g. `RangeAfter(10)`) where
+/// the boundary key (10) is included in the proof as a digest node to anchor
+/// the range, but is not part of the result set.
+pub fn key_exists_as_boundary_in_proof(proof_bytes: &[u8], key: &[u8]) -> Result<bool, Error> {
+    let decoder = Decoder::new(proof_bytes);
+    for op_result in decoder {
+        let op = op_result?;
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply merk/src/proofs/query/verify.rs:687]: it's fine

**Comment by QuantumExplorer** [comment]
Self Reviewed
